schwa/extraction/git_extractor.py

```diff
@@ -145,22 +147,27 @@ def extract_commit(self, hexsha):
 
     def get_new_file_diffs(self, blob):
         diffs_list = [DiffFile(file_b=blob.path, added=True)]
-        if can_parse_file(blob.path) and self.method_granularity:
+        if can_parse_file(blob.path) and (self.method_granularity or self.line_granularity):
             source = GitExtractor.get_source(blob)
             file_parsed = GitExtractor.parse(blob.path, source)
-            if file_parsed:
-                classes_set = file_parsed.get_classes_set()
-                methods_set = file_parsed.get_functions_set()
-                for c in classes_set:
-                    diffs_list.append(DiffClass(file_name=blob.path, class_b=c, added=True))
-                for c, m in methods_set:
-                    diffs_list.append(DiffMethod(file_name=blob.path, class_name=c, method_b=m, added=True))
+            if self.method_granularity:
+                if file_parsed:
+                    classes_set = file_parsed.get_classes_set()
+                    methods_set = file_parsed.get_functions_set()
+                    for c in classes_set:
+                        diffs_list.append(DiffClass(file_name=blob.path, class_b=c, added=True))
+                    for c, m in methods_set:
+                        diffs_list.append(DiffMethod(file_name=blob.path, class_name=c, method_b=m, added=True))
+            if self.line_granularity:
```

Comment:
Line granularity can be independent from Parsing capabilities of the source code. E.g. Although we cannot parse Python, we can compute what lines were changed, added and removed.

Reply:
Good point.  Fixed.